File: virtusbk/src/main/java/com/dua/virtusbk/entity/Person.java
package com.dua.virtusbk.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer;
import lombok.*;
import org.springframework.format.annotation.DateTimeFormat;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import lombok.AllArgsConstructor;

import javax.persistence.*;
import java.time.Instant;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

@Table(name = "persons")
@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
//@RequiredArgsConstructor

public class Person {
    public Person(Long id) {
        this.id = id;
    }

    @Id
    //@GeneratedValue(strategy = GenerationType.AUTO)
    //@GeneratedValue(strategy = GenerationType.SEQUENCE)
    //@GeneratedValue(strategy = GenerationType.SEQUENCE, generator="webuser_idwebuser_seq")
    @SequenceGenerator(name = "persons_id_person_seq", sequenceName = "persons_id_person_seq", allocationSize = 1)
    @GeneratedValue(generator = "persons_id_person_seq")
    @Column(name = "id_person", nullable = false)
    private Long id;

    @Column(name = "name_person", nullable = false, length = 50)
    @Basic(optional = true)
    private String namePerson;

    @Column(name = "lastname_person", nullable = false, length = 50)
    @Basic(optional = true)
    private String lastnamePerson;

    @Column(name = "email_person", nullable = false, length = 75)
    @Basic(optional = true)
    private String emailPerson;

    @Column(name = "password_person", nullable = false, length = 64)
    @Basic(optional = true)
    private String passwordPerson;

    @Column(name = "type_person", nullable = false, length = 1)
    @Basic(optional = true)
    private String typePerson;

    @Column(name = "pathimg_person", length = 200)
    @Basic(optional = true)
    private String pathimgPerson;

    @Column(name = "codeverification_person", length = 15)
    @Basic(optional = true)
    private String codeverificationPerson;

    @Column(name = "datereg_person")
    @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss", iso = DateTimeFormat.ISO.DATE_TIME)
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime dateregPerson;

    @Column(name = "dateupdate_person", nullable = false)
    @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss", iso = DateTimeFormat.ISO.DATE_TIME)
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime dateupdatePerson;

    @Column(name = "provider_person", length = 20)
    @Basic(optional = true)
    private String providerPerson;

    @Column(name = "id_location", length = 20)
    @Basic(optional = true)
    private String idLocation;
}

File: virtus/src/app/models/Person.ts
export class Person {
  private id: number;
  private namePerson: string;
  private lastnamePerson: string;
  private emailPerson: string;
  private passwordPerson: string;
  private typePerson: string;
  private pathimgPerson: string;
  private codeverificationPerson: string;
  private dateregPerson: string;
  private dateupdatePerson: string;
  private providerPerson: string;
  private idLocation: string;


  constructor(id: number, namePerson: string, lastnamePerson: string, emailPerson: string, typePerson: string, pathimgPerson: string, codeverificationPerson: string, dateregPerson: string, dateupdatePerson: string, providerPerson: string, idLocation: string) {
    this.id = id;
    this.namePerson = namePerson;
    this.lastnamePerson = lastnamePerson;
    this.emailPerson = emailPerson;
    this.typePerson = typePerson;
    this.pathimgPerson = pathimgPerson;
    this.codeverificationPerson = codeverificationPerson;
    this.dateregPerson = dateregPerson;
    this.dateupdatePerson = dateupdatePerson;
    this.providerPerson = providerPerson;
    this.idLocation = idLocation;
  }

  get _id(): number {
    return this.id;
  }

  set _id(value: number) {
    this.id = value;
  }

  get _namePerson(): string {
    return this.namePerson;
  }

  set _namePerson(value: string) {
    this.namePerson = value;
  }

  get _lastnamePerson(): string {
    return this.lastnamePerson;
  }

  set _lastnamePerson(value: string) {
    this.lastnamePerson = value;
  }

  get _emailPerson(): string {
    return this.emailPerson;
  }

  set _emailPerson(value: string) {
    this.emailPerson = value;
  }

  get _passwordPerson(): string {
    return this.passwordPerson;
  }

  set _passwordPerson(value: string) {
    this.passwordPerson = value;
  }

  get _typePerson(): string {
    return this.typePerson;
  }

  set _typePerson(value: string) {
    this.typePerson = value;
  }

  get _pathimgPerson(): string {
    return this.pathimgPerson;
  }

  set _pathimgPerson(value: string) {
    this.pathimgPerson = value;
  }

  get _codeverificationPerson(): string {
    return this.codeverificationPerson;
  }

  set _codeverificationPerson(value: string) {
    this.codeverificationPerson = value;
  }

  get _dateregPerson(): string {
    return this.dateregPerson;
  }

  set _dateregPerson(value: string) {
    this.dateregPerson = value;
  }

  get _dateupdatePerson(): string {
    return this.dateupdatePerson;
  }

  set _dateupdatePerson(value: string) {
    this.dateupdatePerson = value;
  }

  get _providerPerson(): string {
    return this.providerPerson;
  }

  set _providerPerson(value: string) {
    this.providerPerson = value;
  }

  get _idLocation(): string {
    return this.idLocation;
  }

  set _idLocation(value: string) {
    this.idLocation = value;
  }
}


